themes/ddbasic/template.php

```diff
@@ -272,20 +272,32 @@ function ddbasic_preprocess_user_picture(&$variables) {
 }
 
 /**
- * Implements hook_preprocess_node().
+ * Implements hook_node_view_alter().
  *
- * Override or insert variables into the node templates.
+ * Add opening hours toggle to opening hours field if present.
  */
-function ddbasic_preprocess_node(&$variables, $hook) {
-  // Opening hours on library list. but not on the search page.
-  $path = drupal_get_path_alias();
-  if (!(strpos($path, 'search', 0) === 0)) {
-    $hooks = theme_get_registry(FALSE);
-    if (isset($hooks['opening_hours_week']) && $variables['type'] == 'ding_library') {
-      $variables['opening_hours'] = theme('opening_hours_week', array('node' => $variables['node']));
+function ddbasic_node_view_alter(&$build) {
+  foreach (array_keys($build) as $field) {
+
+    if (preg_match('/^opening_hours_week_*(\d+)*/', $field, $matches)) {
```

Comment:
`$matches` is undefined here.

Also I would appreciate a comment explaining why using regular expression matching is necessary here. I would expect the field name to be static? Could it be simplified by checking if the field starts with `opening_hours_week_`?